pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -184,11 +188,84 @@ data:
 ---
 {{- end}}
 `
+
+	// vmDeploymentYaml aims to simulate a VM, but instead of managing the complex test setup of spinning up a VM,
+	// connecting, etc we run it inside a pod. The pod has pretty much all Kubernetes features disabled (DNS and SA token mount)
+	// such that we can adequately simulate a VM and DIY the bootstrapping.
+	vmDeploymentYaml = `
+{{- $cluster := .Cluster }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $.Service }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ $.Service }}
+  template:
+    metadata:
+      annotations:
+        # Sidecar is inside the pod to simulate VMs - do not inject
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: {{ $.Service }}
+    spec:
+      # Disable kube-dns, to mirror VM
+      dnsPolicy: Default
+      # Disable service account mount, to mirror VM
+      automountServiceAccountToken: false
+      containers:
+      - name: istio-proxy
+        image: registry:5000/app_sidecar:1590626160
+        #image: {{ $.Hub }}/app_sidecar:{{ $.Tag }}
+        imagePullPolicy: {{ $.PullPolicy }}
+        securityContext:
+          capabilities:
+            add:
+            - NET_ADMIN
+          runAsUser: 1338
+          runAsGroup: 1338
+        command:
+        - bash
+        - -c
+        - |-
+          # Certificates are pre-provisioned in /var/run/secrets/istio
+          sudo mkdir -p /etc/certs
+          sudo cp /var/run/secrets/istio/{root-cert.pem,cert-chain.pem,key.pem} /etc/certs
+          sudo chown -R istio-proxy /etc/certs /var/lib/istio/envoy
+
+          sudo sh -c 'echo ISTIO_SERVICE_CIDR=* > /var/lib/istio/envoy/cluster.env'
+          sudo sh -c 'echo ISTIO_PILOT_PORT={{$.VM.IstiodPort}} >> /var/lib/istio/envoy/cluster.env'
```

Comment:
during the docs review, the decision was made to hardcode this variable and not make it configurable.

Reply:
The docs say a LB is required. In this test we can use nodeport as well, which requires changing the port

Reply:
@howardjohn I am more or less fine with having test cases that test nodeport, although I think the project should be consistent and consistently say "no LB, no Istio". I can't think of a scenario in which nodeport is ever a good choice except in environments where load balancers are not feasible. Given metallb, LBs should be feasible everywhere.